src/segment-loader.js

```diff
@@ -805,185 +720,157 @@ export default class SegmentLoader extends videojs.EventTarget {
-      this.mediaTransferDuration += request.roundTripTime || 0;
+    if (segment.map) {
+      const map = this.initSegments_[initSegmentId(segment.map)];
 
-      if (segment.key) {
-        segmentInfo.encryptedBytes = new Uint8Array(request.response);
+      if (map) {
+        simpleSegment.map = map;
       } else {
-        segmentInfo.bytes = new Uint8Array(request.response);
+        simpleSegment.map = {
+          resolvedUri: segment.map.resolvedUri,
+          byterange: segment.map.byterange
+        };
       }
     }
 
-    if (request === this.xhr_.keyXhr) {
-      // the key request is no longer outstanding
-      this.xhr_.keyXhr = null;
-
-      if (request.response.byteLength !== 16) {
-        this.abort_();
-        this.error({
-          status: request.status,
-          message: 'Invalid HLS key at URL: ' + segment.key.uri,
-          code: 2,
-          xhr: request
-        });
-        this.state = 'READY';
-        this.pause();
-        return this.trigger('error');
-      }
+    return simpleSegment;
+  }
 
-      view = new DataView(request.response);
-      segment.key.bytes = new Uint32Array([
-        view.getUint32(0),
-        view.getUint32(4),
-        view.getUint32(8),
-        view.getUint32(12)
-      ]);
+  /**
+   * Handle the callback from the segmentRequest function and set the
+   * associated SegmentLoader state and errors if necessary
+   *
+   * @private
+   */
+  segmentRequestFinished_(errors, simpleSegment) {
+    // every request counts as a media request even if it has been aborted
+    // or canceled due to a timeout
+    this.mediaRequests += 1;
 
-      // if the media sequence is greater than 2^32, the IV will be incorrect
-      // assuming 10s segments, that would be about 1300 years
-      segment.key.iv = segment.key.iv || new Uint32Array([
-        0, 0, 0, segmentInfo.mediaIndex + segmentInfo.playlist.mediaSequence
-      ]);
+    if (simpleSegment.stats) {
+      this.mediaBytesTransferred += simpleSegment.stats.bytesReceived;
+      this.mediaTransferDuration += simpleSegment.stats.roundTripTime;
     }
 
-    if (request === this.xhr_.initSegmentXhr) {
-      // the init segment request is no longer outstanding
-      this.xhr_.initSegmentXhr = null;
-      segment.map.bytes = new Uint8Array(request.response);
-      this.initSegments_[initSegmentId(segment.map)] = segment.map;
+    // The request was aborted and the SegmentLoader has already been reset
+    if (!this.pendingSegment_) {
+      this.mediaRequestsAborted += 1;
+      return;
     }
 
-    if (!this.xhr_.segmentXhr && !this.xhr_.keyXhr && !this.xhr_.initSegmentXhr) {
-      this.xhr_ = null;
-      this.processResponse_();
+    // the request was aborted and the SegmentLoader has already started
+    // another request. this can happen when the timeout for an aborted
+    // request triggers due to a limitation in the XHR library
+    // do not count this as any sort of request or we risk double-counting
```

Comment:
This is a bit confusing. Can you explain the limitation? (Not requesting you add more detail to the actual code comment)